src/components/item-display-panel/item-display-panel.component.ts

```diff
@@ -0,0 +1,75 @@
+import { Component, Directive, Input, SimpleChange } from '@angular/core';
+import { ModalModule } from 'ngx-bootstrap/modal';
+import { Observable } from 'rxjs/Observable';
+
+@Component({
+    selector: 'ux-item-display-panel',
+    templateUrl: './item-display-panel.component.html',
+    exportAs: 'ux-item-display-panel',
+    host: {
+        '(document:click)': 'clickOff($event)',
+        '(document:keyup.escape)': 'hide()'
+    }
+})
+export class ItemDisplayPanelComponent { 
+    @Input() top: number;
+    @Input() shadow: boolean;
+    @Input() visible: boolean;
+    @Input() title: string;
+    @Input() animate: boolean;
+
+    height: string;
+
+    ngOnChanges(changes: {[top: number]: SimpleChange}) {
+        this.height = 'calc(100% - ' + this.top + 'px)';
+    }
+
+    clickOff(event: any) {
+       
+        // dont do anything if the panel is hidden
+        if (this.visible) {
+            let target = event.target;
+
+            let hidePanel = true;
+
+            while (target.nodeName !== 'BODY') {
```

Comment:
Take a look at the fixes made recently to the AngularJS version: https://github.com/UXAspects/UXAspects/pull/185/files

Include those changes here